### PR TITLE
[Integration Work]: update shim path & nix build

### PR DESCRIPTION
**Pull request by zo-el**

### CODE REVIEW

**Comment by timotree3** [comment]
Oops accidentally committed a bunch of formatting changes. That was my editor's autoformat. Let me know if you want me to revert those

**Comment by alastairong** [comment]
I ran this in a holoport and tried to sign up. Wormhole and lair shim work, but there seems to be resilience issues. In the log below we go through several signing steps and fail at one of them, causing the whole sign up to fail.

```
Feb 12 11:29:36 hpos holo-envoy[9759]: 2021-02-12T11:29:36.831Z [ index.js   ] SILLY: Incoming connection from /hosting/?anonymous=false&agent_id=uhCAkGdlFEZ2ohJzdNlL74U2Ts3V1sRMxMgD6TyFUeOrw9Mim47BY&hha_hash=uhCkkcF0X1dpwHFeIPI6-7rzM6ma9IgyiqD-othxgENSkL1So1Slt
Feb 12 11:29:36 hpos holo-envoy[9759]: 2021-02-12T11:29:36.831Z [ index.js   ] NORMA: Agent (uhCAkGdlFEZ2ohJzdNlL74U2Ts3V1sRMxMgD6TyFUeOrw9Mim47BY) connection for HHA ID: uhCkkcF0X1dpwHFeIPI6-7rzM6ma9IgyiqD-othxgENSkL1So1Slt
Feb 12 11:29:37 hpos holo-envoy[9759]: 2021-02-12T11:29:37.038Z [ index.js   ] SILLY: Incoming websocket message: {"jsonrpc":"2.0","method":"rpc.on","params":["signal"],"id":1}
Feb 12 11:29:37 hpos holo-envoy[9759]: 2021-02-12T11:29:37.039Z [ index.js   ] SILLY: Incoming websocket message: {"jsonrpc":"2.0","method":"holo/wormhole/event","params":["uhCAkGdlFEZ2ohJzdNlL74U2Ts3V1sRMxMgD6TyFUeOrw9Mim47BY"],"id":2}
Feb 12 11:29:37 hpos holo-envoy[9759]: 2021-02-12T11:29:37.039Z [ index.js   ] NORMA: Initializing wormhole setup for Agent (uhCAkGdlFEZ2ohJzdNlL74U2Ts3V1sRMxMgD6TyFUeOrw9Mim47BY)
Feb 12 11:29:37 hpos holo-envoy[9759]: 2021-02-12T11:29:37.039Z [ index.js   ] DEBUG: Registering RPC WebSocket event (uhCAkGdlFEZ2ohJzdNlL74U2Ts3V1sRMxMgD6TyFUeOrw9Mim47BY/wormhole/request) in namespace: /hosting/
Feb 12 11:29:37 hpos holo-envoy[9759]: 2021-02-12T11:29:37.347Z [ index.js   ] SILLY: Incoming websocket message: {"jsonrpc":"2.0","method":"rpc.on","params":["uhCAkGdlFEZ2ohJzdNlL74U2Ts3V1sRMxMgD6TyFUeOrw9Mim47BY/wormhole/request"],"id":3}
Feb 12 11:29:37 hpos holo-envoy[9759]: 2021-02-12T11:29:37.553Z [ index.js   ] SILLY: Incoming websocket message: {"jsonrpc":"2.0","method":"holo/agent/signup","params":["uhCkkcF0X1dpwHFeIPI6-7rzM6ma9IgyiqD-othxgENSkL1So1Slt","uhCAkGdlFEZ2ohJzdNlL74U2Ts3V1sRMxMgD6TyFUeOrw9Mim47BY"],"id":4}
Feb 12 11:29:37 hpos holo-envoy[9759]: 2021-02-12T11:29:37.553Z [ index.js   ] NORMA: Received sign-up request from Agent (uhCAkGdlFEZ2ohJzdNlL74U2Ts3V1sRMxMgD6TyFUeOrw9Mim47BY) for HHA ID: uhCkkcF0X1dpwHFeIPI6-7rzM6ma9IgyiqD-othxgENSkL1So1Slt
Feb 12 11:29:37 hpos holo-envoy[9759]: 2021-02-12T11:29:37.554Z [ index.js   ]  INFO: Retrieve the hosted app cell_data using the anonymous installed_app_id: 'uhCkkcF0X1dpwHFeIPI6-7rzM6ma9IgyiqD-othxgENSkL1So1Slt'
Feb 12 11:29:37 hpos holo-envoy[9759]: 2021-02-12T11:29:37.554Z [ index.js   ] NORMA: Received request to call Conductor using client 'app' with call spec of type 'object'
Feb 12 11:29:37 hpos holo-envoy[9759]: 2021-02-12T11:29:37.554Z [ index.js   ] DEBUG: App Info Call spec details for installed_app_id ( uhCkkcF0X1dpwHFeIPI6-7rzM6ma9IgyiqD-othxgENSkL1So1Slt )
Feb 12 11:29:37 hpos holo-envoy[9759]: 2021-02-12T11:29:37.554Z [ index.js   ] SILLY: Calling Conductor method (appInfo) over client 'app' with input {"installed_app_id":"uhCkkcF0X1dpwHFeIPI6-7rzM6ma9IgyiqD-othxgENSkL1So1Slt"}: 
Feb 12 11:29:37 hpos holo-envoy[9759]: 2021-02-12T11:29:37.556Z [ index.js   ] DEBUG: Successful app interface response: {"installed_app_id":"uhCkkcF0X1dpwHFeIPI6-7rzM6ma9IgyiqD-othxgENSkL1So1Slt","cell_data":[[[{"type":"Buffer","data":[132,45,36,134,132,81,150,23,141,201,221,142,171,216,171,67,117,29,49,119,79,79,100,240,184,56,3,226,17,87,255,180,239,7,124,33,112,8,182]},{"type":"Buffer","data":[132,32,36,117,245,213,242,218,63,12,129,237,38,160,244,213,192,191,199,97,90,110,177,193,230,35,131,71,106,11,186,238,160,48,220,56,95,79,31]}],"elemental-chat"]]} 
Feb 12 11:29:37 hpos holo-envoy[9759]: 2021-02-12T11:29:37.557Z [ index.js   ] NORMA: Conductor call returned successful 'object' response: {"installed_app_id":"uhCkkcF0X1dpwHFeIPI6-7rzM6ma9IgyiqD-othxgENSkL1So1Slt","cell_data":[[[{"type":"Buffer","data":[132,45,36,134,132,81,150,23,141,201,221,142,171,216,171,67,117,29,49,119,79,79,100,240,184,56,3,226,17,87,255,180,239,7,124,33,112,8,182]},{"type":"Buffer","data":[132,32,36,117,245,213,242,218,63,12,129,237,38,160,244,213,192,191,199,97,90,110,177,193,230,35,131,71,106,11,186,238,160,48,220,56,95,79,31]}],"elemental-chat"]]} 
Feb 12 11:29:37 hpos holo-envoy[9759]: 2021-02-12T11:29:37.557Z [ index.js   ] SILLY: NUMBER OF DNAs in the hosted happ: 
Feb 12 11:29:37 hpos holo-envoy[9759]: 2021-02-12T11:29:37.557Z [ index.js   ] SILLY: AppInfo on sign-up: 
Feb 12 11:29:37 hpos holo-envoy[9759]: 2021-02-12T11:29:37.558Z [ index.js   ] SILLY: Hosted App Cell Data: [[[{"type":"Buffer","data":[132,45,36,134,132,81,150,23,141,201,221,142,171,216,171,67,117,29,49,119,79,79,100,240,184,56,3,226,17,87,255,180,239,7,124,33,112,8,182]},{"type":"Buffer","data":[132,32,36,117,245,213,242,218,63,12,129,237,38,160,244,213,192,191,199,97,90,110,177,193,230,35,131,71,106,11,186,238,160,48,220,56,95,79,31]}],"elemental-chat"]]
Feb 12 11:29:37 hpos holo-envoy[9759]: 2021-02-12T11:29:37.558Z [ index.js   ]  INFO: Found 1 DNA(s) for the app bundle with HHA ID: uhCkkcF0X1dpwHFeIPI6-7rzM6ma9IgyiqD-othxgENSkL1So1Slt
Feb 12 11:29:37 hpos holo-envoy[9759]: 2021-02-12T11:29:37.558Z [ index.js   ]  INFO: Encoded Agent ID (uhCAkGdlFEZ2ohJzdNlL74U2Ts3V1sRMxMgD6TyFUeOrw9Mim47BY) into buffer form: {"type":"Buffer","data":[132,32,36,25,217,69,17,157,168,132,156,221,54,82,251,225,77,147,179,117,117,177,19,49,50,0,250,79,33,84,120,234,240,244,200,166,227,176,88]}
Feb 12 11:29:37 hpos holo-envoy[9759]: 2021-02-12T11:29:37.558Z [ index.js   ]  INFO: Installing App with HHA ID (uhCkkcF0X1dpwHFeIPI6-7rzM6ma9IgyiqD-othxgENSkL1So1Slt) as Installed App ID (uhCkkcF0X1dpwHFeIPI6-7rzM6ma9IgyiqD-othxgENSkL1So1Slt:uhCAkGdlFEZ2ohJzdNlL74U2Ts3V1sRMxMgD6TyFUeOrw9Mim47BY) 
Feb 12 11:29:37 hpos holo-envoy[9759]: 2021-02-12T11:29:37.558Z [ index.js   ] DEBUG: installedDnas : [{"nick":"elemental-chat","hash":{"type":"Buffer","data":[132,45,36,134,132,81,150,23,141,201,221,142,171,216,171,67,117,29,49,119,79,79,100,240,184,56,3,226,17,87,255,180,239,7,124,33,112,8,182]}}]
Feb 12 11:29:37 hpos holo-envoy[9759]: 2021-02-12T11:29:37.558Z [ index.js   ] NORMA: Received request to call Conductor using client 'admin' with call spec of type 'string'
Feb 12 11:29:37 hpos holo-envoy[9759]: 2021-02-12T11:29:37.558Z [ index.js   ] DEBUG: Admin Call spec payload: ( installed_app_id : string, agent_key : object, dnas : object )
Feb 12 11:29:37 hpos holo-envoy[9759]: 2021-02-12T11:29:37.559Z [ index.js   ] SILLY: Calling Conductor method (installApp) over client 'admin' with input {"installed_app_id":"uhCkkcF0X1dpwHFeIPI6-7rzM6ma9IgyiqD-othxgENSkL1So1Slt:uhCAkGdlFEZ2ohJzdNlL74U2Ts3V1sRMxMgD6TyFUeOrw9Mim47BY","agent_key":{"type":"Buffer","data":[132,32,36,25,217,69,17,157,168,132,156,221,54,82,251,225,77,147,179,117,117,177,19,49,50,0,250,79,33,84,120,234,240,244,200,166,227,176,88]},"dnas":[{"nick":"elemental-chat","hash":{"type":"Buffer","data":[132,45,36,134,132,81,150,23,141,201,221,142,171,216,171,67,117,29,49,119,79,79,100,240,184,56,3,226,17,87,255,180,239,7,124,33,112,8,182]}}]}: 
Feb 12 11:29:37 hpos holo-envoy[9759]: 2021-02-12T11:29:37.561Z [ wormhole.j ] NORMA: Intercepted sign by public key
Feb 12 11:29:37 hpos holo-envoy[9759]: Wormhole Signing Requested...
Feb 12 11:29:37 hpos holo-envoy[9759]: 2021-02-12T11:29:37.561Z [ index.js   ] NORMA: Opening a request (#41) for Agent (uhCAkGdlFEZ2ohJzdNlL74U2Ts3V1sRMxMgD6TyFUeOrw9Mim47BY) signature of payload: typeof '{"type":"Buffer","data":[132,164,116,121,112,101,163,68,110,97,166,97,117,116,104,111,114,196,39,132,32,36,25,217,69,17,157,168,132,156,221,54,82,251,225,77,147,179,117,117,177,19,49,50,0,250,79,33,84,120,234,240,244,200,166,227,176,88,169,116,105,109,101,115,116,97,109,112,146,206,96,38,102,161,206,33,93,18,52,164,104,97,115,104,196,39,132,45,36,134,132,81,150,23,141,201,221,142,171,216,171,67,117,29,49,119,79,79,100,240,184,56,3,226,17,87,255,180,239,7,124,33,112,8,182]}'
Feb 12 11:29:37 hpos holo-envoy[9759]: 2021-02-12T11:29:37.562Z [ index.js   ]  INFO: Adding signature request #41 to pending signatures
Feb 12 11:29:37 hpos holo-envoy[9759]: 2021-02-12T11:29:37.562Z [ index.js   ] NORMA: Sent signing request #41 to Agent (uhCAkGdlFEZ2ohJzdNlL74U2Ts3V1sRMxMgD6TyFUeOrw9Mim47BY)
Feb 12 11:29:37 hpos holo-envoy[9759]: 2021-02-12T11:29:37.769Z [ index.js   ] SILLY: Incoming websocket message: {"jsonrpc":"2.0","method":"holo/wormhole/response","params":[41,"PCMYibgQCpILOYLbU0DqyNrAANVbx4m9d0S/Ef7NHZVEhKa9xGikVfK++cLRO/xw2f3s4C9SoCS/pc28+2ZbBw=="],"id":5}
Feb 12 11:29:37 hpos holo-envoy[9759]: 2021-02-12T11:29:37.769Z [ index.js   ] NORMA: Received signing response #41 with signature: PCMYibgQCpILOYLbU0DqyNrAANVbx4m9d0S/Ef7NHZVEhKa9xGikVfK++cLRO/xw2f3s4C9SoCS/pc28+2ZbBw==
Feb 12 11:29:37 hpos holo-envoy[9759]: 2021-02-12T11:29:37.773Z [ wormhole.j ] NORMA: Intercepted sign by public key
Feb 12 11:29:37 hpos holo-envoy[9759]: Wormhole Signing Requested...
Feb 12 11:29:37 hpos holo-envoy[9759]: 2021-02-12T11:29:37.774Z [ index.js   ] NORMA: Opening a request (#42) for Agent (uhCAkGdlFEZ2ohJzdNlL74U2Ts3V1sRMxMgD6TyFUeOrw9Mim47BY) signature of payload: typeof '{"type":"Buffer","data":[134,164,116,121,112,101,178,65,103,101,110,116,86,97,108,105,100,97,116,105,111,110,80,107,103,166,97,117,116,104,111,114,196,39,132,32,36,25,217,69,17,157,168,132,156,221,54,82,251,225,77,147,179,117,117,177,19,49,50,0,250,79,33,84,120,234,240,244,200,166,227,176,88,169,116,105,109,101,115,116,97,109,112,146,206,96,38,102,161,206,45,240,219,65,170,104,101,97,100,101,114,95,115,101,113,1,171,112,114,101,118,95,104,101,97,100,101,114,196,39,132,41,36,26,52,153,145,242,55,208,191,174,81,174,116,110,185,161,99,96,3,169,48,210,27,87,72,177,89,96,211,31,30,74,52,78,190,53,91,174,109,101,109,98,114,97,110,101,95,112,114,111,111,102,192]}'
Feb 12 11:29:37 hpos holo-envoy[9759]: 2021-02-12T11:29:37.775Z [ index.js   ]  INFO: Adding signature request #42 to pending signatures
Feb 12 11:29:37 hpos holo-envoy[9759]: 2021-02-12T11:29:37.775Z [ index.js   ] NORMA: Sent signing request #42 to Agent (uhCAkGdlFEZ2ohJzdNlL74U2Ts3V1sRMxMgD6TyFUeOrw9Mim47BY)
Feb 12 11:29:38 hpos holo-envoy[9759]: 2021-02-12T11:29:38.077Z [ index.js   ] SILLY: Incoming websocket message: {"jsonrpc":"2.0","method":"holo/wormhole/response","params":[42,"vUcW4uIyzE0dqWEZeQmS4yrnkdE1XfplkcRUvQOirYJL9vLT74XcxsBdRrVV1a1ijAgYpQ+C1gGl52n5zrzUBQ=="],"id":6}
Feb 12 11:29:38 hpos holo-envoy[9759]: 2021-02-12T11:29:38.078Z [ index.js   ] NORMA: Received signing response #42 with signature: vUcW4uIyzE0dqWEZeQmS4yrnkdE1XfplkcRUvQOirYJL9vLT74XcxsBdRrVV1a1ijAgYpQ+C1gGl52n5zrzUBQ==
Feb 12 11:29:38 hpos holo-envoy[9759]: 2021-02-12T11:29:38.081Z [ wormhole.j ] NORMA: Intercepted sign by public key
Feb 12 11:29:38 hpos holo-envoy[9759]: Wormhole Signing Requested...
Feb 12 11:29:38 hpos holo-envoy[9759]: 2021-02-12T11:29:38.083Z [ index.js   ] NORMA: Opening a request (#43) for Agent (uhCAkGdlFEZ2ohJzdNlL74U2Ts3V1sRMxMgD6TyFUeOrw9Mim47BY) signature of payload: typeof '{"type":"Buffer","data":[135,164,116,121,112,101,166,67,114,101,97,116,101,166,97,117,116,104,111,114,196,39,132,32,36,25,217,69,17,157,168,132,156,221,54,82,251,225,77,147,179,117,117,177,19,49,50,0,250,79,33,84,120,234,240,244,200,166,227,176,88,169,116,105,109,101,115,116,97,109,112,146,206,96,38,102,162,206,4,189,140,103,170,104,101,97,100,101,114,95,115,101,113,2,171,112,114,101,118,95,104,101,97,100,101,114,196,39,132,41,36,240,123,183,126,241,168,192,22,11,229,43,104,29,226,225,134,114,93,236,236,138,73,24,152,251,230,21,74,176,234,93,161,123,222,42,237,170,101,110,116,114,121,95,116,121,112,101,129,171,65,103,101,110,116,80,117,98,75,101,121,192,170,101,110,116,114,121,95,104,97,115,104,196,39,132,33,36,25,217,69,17,157,168,132,156,221,54,82,251,225,77,147,179,117,117,177,19,49,50,0,250,79,33,84,120,234,240,244,200,166,227,176,88]}'
Feb 12 11:29:38 hpos holo-envoy[9759]: 2021-02-12T11:29:38.083Z [ index.js   ]  INFO: Adding signature request #43 to pending signatures
Feb 12 11:29:38 hpos holo-envoy[9759]: 2021-02-12T11:29:38.084Z [ index.js   ] NORMA: Sent signing request #43 to Agent (uhCAkGdlFEZ2ohJzdNlL74U2Ts3V1sRMxMgD6TyFUeOrw9Mim47BY)
Feb 12 11:29:38 hpos holo-envoy[9759]: 2021-02-12T11:29:38.393Z [ index.js   ] SILLY: Incoming websocket message: {"jsonrpc":"2.0","method":"holo/wormhole/response","params":[43,"7lHju92aW6OFfW0eT1FEFcxX/QmyyYBRgeX9E5149LuYIXF9zZxgWhE4z8aLSt/Sr5nQEPw+JCGD5fwl9GJWBA=="],"id":7}
Feb 12 11:29:38 hpos holo-envoy[9759]: 2021-02-12T11:29:38.394Z [ index.js   ] NORMA: Received signing response #43 with signature: 7lHju92aW6OFfW0eT1FEFcxX/QmyyYBRgeX9E5149LuYIXF9zZxgWhE4z8aLSt/Sr5nQEPw+JCGD5fwl9GJWBA==
Feb 12 11:29:38 hpos holo-envoy[9759]: 2021-02-12T11:29:38.396Z [ index.js   ] DEBUG: Successful admin interface response: {"installed_app_id":"uhCkkcF0X1dpwHFeIPI6-7rzM6ma9IgyiqD-othxgENSkL1So1Slt:uhCAkGdlFEZ2ohJzdNlL74U2Ts3V1sRMxMgD6TyFUeOrw9Mim47BY","cell_data":[[[{"type":"Buffer","data":[132,45,36,134,132,81,150,23,141,201,221,142,171,216,171,67,117,29,49,119,79,79,100,240,184,56,3,226,17,87,255,180,239,7,124,33,112,8,182]},{"type":"Buffer","data":[132,32,36,25,217,69,17,157,168,132,156,221,54,82,251,225,77,147,179,117,117,177,19,49,50,0,250,79,33,84,120,234,240,244,200,166,227,176,88]}],"elemental-chat"]],"type":"success"} 
Feb 12 11:29:38 hpos holo-envoy[9759]: 2021-02-12T11:29:38.397Z [ index.js   ] NORMA: Conductor call returned successful 'object' response: {"installed_app_id":"uhCkkcF0X1dpwHFeIPI6-7rzM6ma9IgyiqD-othxgENSkL1So1Slt:uhCAkGdlFEZ2ohJzdNlL74U2Ts3V1sRMxMgD6TyFUeOrw9Mim47BY","cell_data":[[[{"type":"Buffer","data":[132,45,36,134,132,81,150,23,141,201,221,142,171,216,171,67,117,29,49,119,79,79,100,240,184,56,3,226,17,87,255,180,239,7,124,33,112,8,182]},{"type":"Buffer","data":[132,32,36,25,217,69,17,157,168,132,156,221,54,82,251,225,77,147,179,117,117,177,19,49,50,0,250,79,33,84,120,234,240,244,200,166,227,176,88]}],"elemental-chat"]],"type":"success"} 
Feb 12 11:29:38 hpos holo-envoy[9759]: 2021-02-12T11:29:38.398Z [ index.js   ]  INFO: Activating Installed App (uhCkkcF0X1dpwHFeIPI6-7rzM6ma9IgyiqD-othxgENSkL1So1Slt:uhCAkGdlFEZ2ohJzdNlL74U2Ts3V1sRMxMgD6TyFUeOrw9Mim47BY)
Feb 12 11:29:38 hpos holo-envoy[9759]: 2021-02-12T11:29:38.398Z [ index.js   ] NORMA: Received request to call Conductor using client 'admin' with call spec of type 'string'
Feb 12 11:29:38 hpos holo-envoy[9759]: 2021-02-12T11:29:38.399Z [ index.js   ] DEBUG: Admin Call spec payload: ( installed_app_id : string )
Feb 12 11:29:38 hpos holo-envoy[9759]: 2021-02-12T11:29:38.399Z [ index.js   ] SILLY: Calling Conductor method (activateApp) over client 'admin' with input {"installed_app_id":"uhCkkcF0X1dpwHFeIPI6-7rzM6ma9IgyiqD-othxgENSkL1So1Slt:uhCAkGdlFEZ2ohJzdNlL74U2Ts3V1sRMxMgD6TyFUeOrw9Mim47BY"}: 
Feb 12 11:29:38 hpos holo-envoy[9759]: 2021-02-12T11:29:38.401Z [ wormhole.j ] NORMA: Intercepted sign by public key
Feb 12 11:29:38 hpos holo-envoy[9759]: Wormhole Signing Requested...
Feb 12 11:29:38 hpos holo-envoy[9759]: 2021-02-12T11:29:38.401Z [ index.js   ] NORMA: Opening a request (#44) for Agent (uhCAkw8dEl7lzeRsDQzQ3Lhjcqhf6Mo0SsI7KxKbOE5nko9AOQQSY) signature of payload: typeof '{"type":"Buffer","data":[133,165,115,112,97,99,101,196,36,134,132,81,150,23,141,201,221,142,171,216,171,67,117,29,49,119,79,79,100,240,184,56,3,226,17,87,255,180,239,7,124,33,112,8,182,165,97,103,101,110,116,196,36,195,199,68,151,185,115,121,27,3,67,52,55,46,24,220,170,23,250,50,141,18,176,142,202,196,166,206,19,153,228,163,208,14,65,4,152,164,117,114,108,115,145,217,97,107,105,116,115,117,110,101,45,112,114,111,120,121,58,47,47,116,102,51,83,72,81,66,107,74,99,76,86,100,112,83,69,74,72,99,66,114,78,97,71,95,84,83,49,73,81,72,103,104,50,102,105,55,86,103,81,87,81,77,47,107,105,116,115,117,110,101,45,113,117,105,99,47,104,47,49,54,53,46,50,50,46,51,50,46,49,49,47,112,47,53,55,55,56,47,45,45,172,115,105,103,110,101,100,95,97,116,95,109,115,207,0,0,1,119,150,0,234,95,176,101,120,112,105,114,101,115,95,97,102,116,101,114,95,109,115,206,0,27,119,64]}'
Feb 12 11:29:38 hpos holo-envoy[9759]: 2021-02-12T11:29:38.402Z [ index.js   ]  INFO: Adding signature request #44 to pending signatures
Feb 12 11:29:38 hpos holo-envoy[9759]: 2021-02-12T11:29:38.402Z [ index.js   ] NORMA: Sent signing request #44 to Agent (uhCAkw8dEl7lzeRsDQzQ3Lhjcqhf6Mo0SsI7KxKbOE5nko9AOQQSY)
Feb 12 11:29:43 hpos holo-envoy[9759]: 2021-02-12T11:29:43.406Z [ index.js   ] ERROR: Failed during signing request #44 with timeout (5000ms)
Feb 12 11:29:43 hpos holo-envoy[9759]: (node:9759) UnhandledPromiseRejectionWarning: Failed to get signature from Chaperone
Feb 12 11:29:43 hpos holo-envoy[9759]: (node:9759) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 2)
Feb 12 11:29:43 hpos holo-envoy[9759]: (node:9759) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
Feb 12 11:29:53 hpos holo-envoy[9759]: 2021-02-12T11:29:53.410Z [ index.js   ] ERROR: Failed during Conductor call with error: Error: CONDUCTOR CALL ERROR: {}
Feb 12 11:29:53 hpos holo-envoy[9759]: 2021-02-12T11:29:53.410Z [ index.js   ] ERROR: Failed during 'activateApp': HoloError: Error: CONDUCTOR CALL ERROR: {}
Feb 12 11:29:53 hpos holo-envoy[9759]: 2021-02-12T11:29:53.410Z [ index.js   ] ERROR: Failed during DNA processing for Agent (uhCAkGdlFEZ2ohJzdNlL74U2Ts3V1sRMxMgD6TyFUeOrw9Mim47BY) HHA ID (uhCkkcF0X1dpwHFeIPI6-7rzM6ma9IgyiqD-othxgENSkL1So1Slt): HoloError: Error: CONDUCTOR CALL ERROR: {}
Feb 12 11:29:53 hpos holo-envoy[9759]: 2021-02-12T11:29:53.410Z [ index.js   ] ERROR: Failed during sign-up process for Agent (uhCAkGdlFEZ2ohJzdNlL74U2Ts3V1sRMxMgD6TyFUeOrw9Mim47BY) HHA ID (uhCkkcF0X1dpwHFeIPI6-7rzM6ma9IgyiqD-othxgENSkL1So1Slt): {"name":"HoloError","message":"Failed to create a new hosted agent"}
Feb 12 11:29:53 hpos holo-envoy[9759]: 2021-02-12T11:29:53.728Z [ index.js   ] NORMA: Socket is closing for Agent (uhCAkGdlFEZ2ohJzdNlL74U2Ts3V1sRMxMgD6TyFUeOrw9Mim47BY) using HHA ID uhCkkcF0X1dpwHFeIPI6-7rzM6ma9IgyiqD-othxgENSkL1So1Slt
```

**Comment by zo-el** [comment]
> Failed during signing request

It looks like we may need to increase the timeout.

File: tests/e2e/test_chaperone.js [hunk 1]
```diff
@@ -234,16 +234,40 @@ describe("Server", () => {
...
-      }, host_agent_id, registered_agent, REGISTERED_HAPP_HASH);
+
+        function delay(t, val) {
+          return new Promise(function(resolve) {
+            setTimeout(function() {
+              resolve(val);
+            }, t);
+          });
+        }
+        await delay(10000);
```
Comment (timotree3) [inline tests/e2e/test_chaperone.js:256]: What is the point of this delay? Maybe add a comment explaining

**Review by timotree3 — APPROVED** [review]
Left one question. Otherwise looks good

**Pull request merged**